src/main/java/apoc/index/IndexUpdateTransactionEventHandler.java

```diff
@@ -0,0 +1,114 @@
+package apoc.index;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.event.TransactionData;
+import org.neo4j.graphdb.event.TransactionEventHandler;
+import org.neo4j.graphdb.index.Index;
+import org.neo4j.graphdb.index.IndexManager;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+
+import java.util.*;
+
+/**
+ * a transaction event handler that updates manual indexes based on configuration in graph properties
+ * @author Stefan Armbruster
+ */
+public class IndexUpdateTransactionEventHandler extends TransactionEventHandler.Adapter<Void> {
+
+    private final GraphDatabaseService graphDatabaseService;
+    private final Log log;
+
+    /** maintain a transposed version of configuration of all legacy indexes */
+    private Map<String, Map<String, Index<Node>>> propertyNameAndLabelMatrix = null;
+
+    public IndexUpdateTransactionEventHandler(GraphDatabaseAPI graphDatabaseService, Log log) {
+        this.graphDatabaseService = graphDatabaseService;
+        this.log = log;
+    }
+
+    @Override
+    public Void beforeCommit(TransactionData data) throws Exception {
+
+        data.assignedNodeProperties().forEach(nodePropertyEntry -> {
+            Map<String, Index<Node>> labelToIndexes = getPropertyNameAndLabelMatrix().get(nodePropertyEntry.key());
+            if (labelToIndexes != null) {
+                Node entity = nodePropertyEntry.entity();
+                entity.getLabels().forEach(label -> {
+                            String labelName = label.name();
+                            Index<Node> index = labelToIndexes.get(labelName);
+                            if (index != null) {
+                                String indexKey = labelName + ":" + nodePropertyEntry.key();
+                                Object value = nodePropertyEntry.value();
+                                if (nodePropertyEntry.previouslyCommitedValue()!=null) {
+                                    index.remove(entity, indexKey);
+                                    index.remove(entity, FreeTextSearch.KEY);
```

Comment:
could it be that we remove too many values here? As in this index (afaik) it stores multiple value entries per key, there is also a remove method that could take the previouslyCommittedValue()